presto-raptor/src/test/java/com/facebook/presto/raptor/TestRaptorDistributedQueries.java

```diff
@@ -145,4 +150,86 @@ public void testCreateBucketedTable()
+                .filter(row -> ((String) row.getField(1)).startsWith("system_tables_"))
+                .collect(ImmutableCollectors.toImmutableMap(row -> ((String) row.getField(1))));
+        assertEquals(map.size(), 4);
+        assertEquals(
+                map.get("system_tables_none").getFields(),
+                asList("tpch", "system_tables_none", "t1", ImmutableList.of(), null, 0L, ImmutableList.of()));
+        assertEquals(
+                map.get("system_tables_ordered").getFields(),
+                asList("tpch", "system_tables_ordered", "t3", ImmutableList.of("o3", "o4"), null, 0L, ImmutableList.of()));
+        assertEquals(
+                map.get("system_tables_temporal_bucketed").getFields(),
+                asList("tpch", "system_tables_temporal_bucketed", "t5", ImmutableList.of(), null, 40L, ImmutableList.of("b5", "b6")));
+        assertEquals(
+                map.get("system_tables_temporal_ordered_bucketed").getFields(),
+                asList("tpch", "system_tables_temporal_ordered_bucketed", "t7", ImmutableList.of("o7", "o8"), "test_distribution", 50L, ImmutableList.of("b7", "b8")));
+
+        actualResults = computeActual("SELECT * FROM system.tables WHERE table_schema='tpch'");
+        assertEquals(actualResults.getTypes(), ImmutableList.of(VARCHAR, VARCHAR, VARCHAR, new ArrayType(VARCHAR), VARCHAR, BIGINT, new ArrayType(VARCHAR)));
+        map = actualResults.getMaterializedRows().stream()
+                .filter(row -> ((String) row.getField(1)).startsWith("system_tables_"))
+                .collect(ImmutableCollectors.toImmutableMap(row -> ((String) row.getField(1))));
+        assertEquals(map.size(), 4);
+        assertEquals(
+                map.get("system_tables_none").getFields(),
+                asList("tpch", "system_tables_none", "t1", ImmutableList.of(), null, 0L, ImmutableList.of()));
+        assertEquals(
+                map.get("system_tables_ordered").getFields(),
+                asList("tpch", "system_tables_ordered", "t3", ImmutableList.of("o3", "o4"), null, 0L, ImmutableList.of()));
+        assertEquals(
+                map.get("system_tables_temporal_bucketed").getFields(),
+                asList("tpch", "system_tables_temporal_bucketed", "t5", ImmutableList.of(), null, 40L, ImmutableList.of("b5", "b6")));
+        assertEquals(
+                map.get("system_tables_temporal_ordered_bucketed").getFields(),
+                asList("tpch", "system_tables_temporal_ordered_bucketed", "t7", ImmutableList.of("o7", "o8"), "test_distribution", 50L, ImmutableList.of("b7", "b8")));
+
+        actualResults = computeActual("SELECT * FROM system.tables WHERE table_name='system_tables_temporal_ordered_bucketed'");
+        assertEquals(actualResults.getTypes(), ImmutableList.of(VARCHAR, VARCHAR, VARCHAR, new ArrayType(VARCHAR), VARCHAR, BIGINT, new ArrayType(VARCHAR)));
+        map = actualResults.getMaterializedRows().stream()
```

Comment:
For these filter tests, I think we just assert the size of the result set, rather than repeating everything
```
assertEquals(actualResults. getMaterializedRows().size(), 1);